tilequeue/command.py

```diff
@@ -962,43 +962,84 @@ def tilequeue_prune_tiles_of_interest(cfg, peripherals):
     assert redshift_days_to_query, ("Number of days to query "
                                     "redshift is not specified")
 
+    redshift_zoom_cutoff = int(redshift_cfg.get('max-zoom', '16'))
+
     s3_parts = prune_cfg.get('s3')
     assert s3_parts, ("The name of an S3 bucket containing tiles "
                       "to delete must be specified")
 
-    new_toi = set()
+    redshift_results = dict()
     with psycopg2.connect(redshift_uri) as conn:
         with conn.cursor() as cur:
             cur.execute("""
-                select x, y, z, tilesize
+                select x, y, z, tilesize, count(*)
                 from tile_traffic_v4
                 where (date >= dateadd(day, -{days}, current_date))
-                  and (z between 0 and 15)
+                  and (z between 0 and {max_zoom})
                   and (x between 0 and pow(2,z)-1)
                   and (y between 0 and pow(2,z)-1)
                 group by z, x, y, tilesize
                 order by z, x, y, tilesize
-                """.format(days=redshift_days_to_query))
-            n_trr = cur.rowcount
-            for (x, y, z, tile_size) in cur:
+                """.format(
+                    days=redshift_days_to_query,
+                    max_zoom=redshift_zoom_cutoff,
+            ))
+            for (x, y, z, tile_size, count) in cur:
                 coord = create_coord(x, y, z)
                 coord_int = coord_marshall_int(coord)
 
                 if not tile_size or tile_size == '256':
                     # "uplift" a tile that is not explicitly a '512' size tile
                     coord_int = coord_int_zoom_up(coord_int)
 
-                new_toi.add(coord_int)
+                # Sum the counts from the 256 and 512 tile requests into the
+                # slot for the 512 tile.
+                existing_count = redshift_results.get(coord_int)
+                if existing_count:
+                    existing_count += count
+                else:
+                    existing_count = count
+
+                redshift_results[coord_int] = existing_count
 
-    logger.info('Fetching tiles recently requested ... done. %s found', n_trr)
+    logger.info('Fetching tiles recently requested ... done. %s found',
+                len(redshift_results))
+
+    cutoff_cfg = prune_cfg.get('cutoff', {})
+    cutoff_requests = cutoff_cfg.get('min-requests', 0)
+    cutoff_tiles = cutoff_cfg.get('max-tiles', 0)
+
+    logger.info('Finding %s tiles requested %s+ times ...',
+                cutoff_tiles,
+                cutoff_requests,
+                )
+
+    # Sort redshift_results (a dict of coord_int->score) by the score value,
+    # filter out the coords that didn't meet the requests cutoff,
+    # then take the first `cutoff_tiles`.
+
+    new_toi = set(
+        t[0] for t in filter(
+            lambda t: t[1] >= cutoff_requests,
+            sorted(redshift_results.iteritems(), key=lambda t: t[1])
+        )[:cutoff_tiles]
```

Comment:
Do you think this would be easier to read if it were spelled out more? Something like: (untested)

```python
new_toi = set()
for coord_int, score in sorted(redshift_results.iteritems(), key=operator.itemgetter(1), reverse=True):
    if score >= cutoff_requests:
        new_toi.add(coord_int)
        if len(new_toi) >= cutoff_tiles:
            break
```